ern-core/src/MiniApp.js

```diff
@@ -267,14 +271,14 @@ Are you sure this is a MiniApp ?`)
         await spin(`${versionLessDependency.toString()} is not declared in the manifest. Performing additional checks.`,
                     yarn.add(DependencyPath.fromString(dependency.toString())))
 
-        const nativeDependencies = findNativeDependencies(path.join(tmpPath, 'node_modules'))
-        if (nativeDependencies.length === 0) {
+        const nativeDependencies = await nativeDependenciesLookup.findNativeDependencies(path.join(tmpPath, 'node_modules'))
+        if (nativeDependencies.all.length === 0) {
           log.debug('Pure JS dependency')
           // This is a pure JS dependency. Not much to do here -yet-
           finalDependency = versionLessDependency
-        } else if (nativeDependencies.length >= 1) {
-          log.debug(`One or more native dependencies identified: ${JSON.stringify(nativeDependencies)}`)
-          for (let dep: Dependency of nativeDependencies) {
+        } else if (nativeDependencies.all.length >= 1) {
```

Comment:
if nativeDependencies is never going to be null. it is okay to leave it unguarded . Otherwise a check would guard it. same as above

Reply:
yes it's never going to be null/undefined, empty array by default, but will use `_.isEmpty` instead, as it is more proper

Reply:
yes it's never going to be null/undefined